### PR TITLE
Fix ReplyView when its content has been redacted.

### LINKED ISSUE #2396: Replies to redactions don’t render sensibly
### Steps to reproduce

1. reply to a msg
2. redact the original msg
3. reply ends up looking like this:

image

#### What did you expect?

redacted msgs are shown as “deleted message” or something rather than just showing blank, which makes it look like the reply has failed to load.

### Your phone model

_No response_

### Application version

508

### Will you send logs?

No

### PR DESCRIPTION
**Pull request by pixlwave**
This PR adds a `redacted` type for reply content. Fixes #2396.

Simulator Screenshot - iPhone 15 Pro - 2024-03-26 at 09 22 25 copy

### CODE REVIEW

File: ElementX/Sources/Screens/RoomScreen/View/Replies/TimelineReplyView.swift [hunk 1]
```diff
@@ -305,7 +314,9 @@ struct TimelineReplyView_Previews: PreviewProvider, TestablePreview {
                 previewItems[index]
             }
         }
+        .padding()
         .environmentObject(viewModel.context)
         .snapshot(delay: 0.2) // Allow member names to load.
+        .previewLayout(.sizeThatFits)
```
Comment (stefanceriu) [inline ElementX/Sources/Screens/RoomScreen/View/Replies/TimelineReplyView.swift:320]: Are you sure this work? 😁

**Review by stefanceriu — APPROVED** [review]
Yey! 👏

File: ElementX/Sources/Screens/RoomScreen/View/Timeline/RedactedRoomTimelineView.swift [hunk 2]
```diff
@@ -22,7 +22,7 @@ struct RedactedRoomTimelineView: View {
     
     var body: some View {
         TimelineStyler(timelineItem: timelineItem) {
-            Label(timelineItem.body, systemImage: "trash")
+            Label(timelineItem.body, icon: \.delete, iconSize: .small, relativeTo: .compound.bodyLG)
```
Comment (pixlwave) [inline ElementX/Sources/Screens/RoomScreen/View/Timeline/RedactedRoomTimelineView.swift:25]: I fixed this icon too, as it must have been missed in the migration from SF Symbols to Compound.

File: ElementX/Sources/Screens/RoomScreen/View/Replies/TimelineReplyView.swift (hunk 1 above)
Comment (pixlwave) [reply ElementX/Sources/Screens/RoomScreen/View/Replies/TimelineReplyView.swift:320]: lol its good enough for me right now.

Comment (stefanceriu) [reply ElementX/Sources/Screens/RoomScreen/View/Replies/TimelineReplyView.swift:320]: 😅

**Comment by bmarty** [comment]
FWIW this is how reply of deleted Event renders on EXA:

image

**Pull request merged**